### PR TITLE
Fix bug 2, 3 and 4 and create constants for common values in the code

### PR DESCRIPTION
**Pull request by DavidGhergut**
Partially solves issues #33 and #34

### CODE REVIEW

File: src/EPS.cpp [hunk 1]
```diff
@@ -3,6 +3,29 @@
 //
 #include "EPS.h"
 
+// Constants for command codes and other identifiers
+const uint8_t STID = 0x00;
```
Comment (ViciousDoormat) [inline src/EPS.cpp]: Mention this bypasses the STID check (page 17)

File: src/EPS.cpp [hunk 2]
```diff
@@ -3,6 +3,29 @@
...
+const uint8_t IVID = 0x06;
```
Comment (ViciousDoormat) [inline src/EPS.cpp]: mention this is required according to page 19

File: src/EPS.cpp [hunk 3]
```diff
@@ -3,6 +3,29 @@
...
+const uint8_t BID = 0x00;
```
Comment (ViciousDoormat) [inline src/EPS.cpp]: Mention this bypasses the BID (page 20)

File: src/EPS.cpp [hunk 4]
```diff
@@ -86,10 +110,10 @@ EPS::config_reply EPS::set_config_params(DWire &wire, uint8_t i2c_address, uint8
     wire.beginTransmission(i2c_address);
 
     // Command structure based on page 63
-    wire.write(0x00); // STID
-    wire.write(0x06); // IVID
-    wire.write(0x84); // Command code (Reset Configuration Parameter)
-    wire.write(0x00); // BID
+    wire.write(STID); // STID
```
Comment (ViciousDoormat) [inline src/EPS.cpp]: As the constants now capture what they mean, the comments are no longer needed. This also refers to all other lines

File: src/EPS.cpp [hunk 5]
```diff
@@ -3,6 +3,29 @@
...
+const uint8_t CMD_CODE_GET_PBU_HOUSEKEEPING_DATA_RAW = 0x60;    // reference: page 53 of 87 (ICD)
+const uint8_t CMD_CODE_GET_PCU_HOUSEKEEPING_DATA_ENG = 0x72;    // reference: page 60 of 87 (ICD)
+const uint8_t CMD_CODE_GET_PCU_HOUSEKEEPING_DATA_RAW = 0x70;    // reference: page 58 of 87 (ICD)
+const uint8_t CMD_CODE_SWITCH_NOMINAL_MODE = 0x30;              // reference: page 38 of 87 (ICD)
+const uint8_t CMD_CODE_OUTPUT_BUS_CHANNEL_OFF = 0x18;           // reference: page 37 of 87 (ICD)
+const uint8_t CMD_CODE_OUTPUT_BUS_GROUP_STATE = 0x14;           // reference: page 35 of 87 (ICD)
+const uint8_t CMD_CODE_OUTPUT_BUS_GROUP_OFF = 0x12;             // reference: page 34 of 87 (ICD)
+const uint8_t CMD_CODE_OUTPUT_BUS_GROUP_ON = 0x10;              // reference: page 33 of 87 (ICD)
+const uint8_t CMD_CODE_RESET_CONFIGURATION = 0x90;              // reference: page 66 of 87 (ICD)
+const uint8_t CONF_KEY_RESET_CONFIGURATION = 0xA7;              // reference: page 66 of 87 (ICD)
```
Comment (mavaneerden) [inline src/EPS.cpp]: Use enums to categorize these constants.

File: src/EPS.cpp [hunk 6]
```diff
@@ -114,11 +138,11 @@ EPS::config_reply EPS::set_config_params(DWire &wire, uint8_t i2c_address, uint8
 
     // If the response is the expected length, process the reply
     if (response == response_length) {
-        reply.stid = wire.read();  // STID
-        reply.ivid = wire.read();  // IVID
-        reply.rc = wire.read();    // Response code
-        reply.bid = wire.read();   // BID
-        reply.stat = wire.read();  // Status byte
```
Comment (mavaneerden) [inline src/EPS.cpp:121]: Why comments removed? I think they are valuable.

**Review by mavaneerden — CHANGES_REQUESTED** [review]

File: src/EPS.cpp (hunk 6 above)
Comment (ViciousDoormat) [reply src/EPS.cpp:121]: These ones could still be somewhat valuable, but the variable names written to already cover them for the most part. In my opinion, the documentation of reply should cover the meaning of its variables (like stat = status code), not the comments in another code file.

Comment (DavidGhergut) [reply src/EPS.cpp:121]: I have readded them in the latest commit! Even though I can also argue they are quite redundant, maybe for a person that hasn't seen the code before, it is better to point out what every attribute means

File: src/EPS.cpp (hunk 5 above)
Comment (DavidGhergut) [reply src/EPS.cpp]: @mavaneerden It should be done in the last commits!

File: src/EPS.cpp [hunk 7]
```diff
@@ -147,11 +147,11 @@ EPS::config_reply EPS::set_config_params(DWire &wire, uint8_t i2c_address, uint8
 
     // If the response is the expected length, process the reply
     if (response == response_length) {
-        reply.stid = wire.read();
-        reply.ivid = wire.read();
-        reply.rc = wire.read();
-        reply.bid = wire.read();
-        reply.stat = wire.read();
+        reply.stid = wire.read();   // STID
```
Comment (ViciousDoormat) [inline src/EPS.cpp:150]: This could also be made into a function to witch you pass "reply" right? As it is repeated multiple times

Comment (DavidGhergut) [reply src/EPS.cpp:150]: Yes, I will definitely do that in a bit!

Comment (DavidGhergut) [reply src/EPS.cpp:150]: Nevermind, I realized that I need to change a bit more if a want a function for the reply, thus I think I will do it tomorrow!

**Comment by ViciousDoormat** [comment]
Standard reading logic has not yet been abstracted towards a function, but merging has been performed to already start adding functionality to refactored code.

**Pull request merged**